In [1]:
import pandas as pd
import numpy as np
import math
#df = pd.read_csv('data/titanic passenger list.csv')

# Statistical indicators

Both ```numpy``` and ```pandas``` allow to easly compute some statistical indicators

## with ```numpy```

In [2]:
dice=np.array([1, 2, 3, 4, 5, 6])

In [3]:
print("mean:", dice.mean())

mean: 3.5


In [4]:
print("variance:", dice.var())

variance: 2.9166666666666665


In [5]:
print("std. dev.:", dice.std())

std. dev.: 1.707825127659933


## with ```pandas```

In [6]:
df = pd.read_csv('data.csv')

In [7]:
df.head()

Duration  Pulse  Maxpulse  Calories
0        60    110       130     409.1
1        60    117       145     479.0
2        60    103       135     340.0
3        45    109       175     282.4
4        45    117       148     406.0

In [8]:
df.describe()

Duration       Pulse    Maxpulse     Calories
count  169.000000  169.000000  169.000000   164.000000
mean    63.846154  107.461538  134.047337   375.790244
std     42.299949   14.510259   16.450434   266.379919
min     15.000000   80.000000  100.000000    50.300000
25%     45.000000  100.000000  124.000000   250.925000
50%     60.000000  105.000000  131.000000   318.600000
75%     60.000000  111.000000  141.000000   387.600000
max    300.000000  159.000000  184.000000  1860.400000

### Mean value

In [10]:
df['Duration'].mean()

63.84615384615385

In [10]:
(df['Duration']**2).mean()

5855.029585798817

### Median



In [11]:
df['Duration'].median()

60.0

### Mode



In [12]:
df['Duration'].mode()

0    60
Name: Duration, dtype: int64

In caso di più valori con la stessa frequenza massima, saranno tutti riprotati. Di seguito, si estrae il primo valore.

In [13]:
df['Duration'].mode().values[0]

60

## standard deviation




In [14]:
df['Duration'].std()

42.299949341408286

In [15]:
math.sqrt((df['Duration']**2).mean()-(df['Duration'].mean())**2)

42.174615882685536

In [16]:
df['Duration'].std(ddof=0)

42.17461588268545

In [17]:
for col in df.columns:
    print(f'std({col}): ', df[col].std())

std(Duration):  42.299949341408286
std(Pulse):  14.510259096501994
std(Maxpulse):  16.450434446024435
std(Calories):  266.37991924435164


## other indicators


In [18]:
df['Duration'].skew()

2.8638875018047085

In [19]:
df['Duration'].min()

15

In [20]:
df['Duration'].max()

300

In [21]:
df

Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
..        ...    ...       ...       ...
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

[169 rows x 4 columns]

In [22]:
df.corr()

Duration     Pulse  Maxpulse  Calories
Duration  1.000000 -0.155408  0.009403  0.922717
Pulse    -0.155408  1.000000  0.786535  0.025121
Maxpulse  0.009403  0.786535  1.000000  0.203813
Calories  0.922717  0.025121  0.203813  1.000000

"Duration" and "Calories" got a 0.922721 correlation, which is a very good correlation, and we can predict that the longer you work out, the more calories you burn, and the other way around: if you burned a lot of calories, you probably had a long work out.

"Duration" and "Maxpulse" got a 0.009403 correlation, which is a very bad correlation, meaning that we can not predict the max pulse by just looking at the duration of the work out, and vice versa.